model/SearchEngine/Driver/Elasticsearch/ElasticSearchClientFactory.php

```diff
@@ -48,6 +48,7 @@ public function create(): Client
 
         return ClientBuilder::create()
                 ->setHosts($this->config->getHosts())
+                ->setBasicAuthentication($this->config->getUsername(), $this->config->getPassword())
```

Comment:
This was already supported as part of the hosts we have the username and password https://github.com/oat-sa/extension-tao-advanced-search/blob/master/README.md#activate-with-credentials-needed

Reply:
->setHosts() Doesnt support array for a single host. It should contain only host string now. Because of that we need to set credentials separately to the builder

Reply:
Thanks for checking!